### ISSUE TITLE
[rock-dkms-bin] Error downloading sources

### ISSUE THREAD

**Issue by vsauer** [rocm-arch/rocm-arch#286]
```
==> Making package: rock-dkms-firmware-bin 3.5-2 (Mon 29 Jun 2020 05:42:50 PM PDT)
==> Retrieving sources...
  -> Downloading rock-dkms-firmware_3.5-32_all.deb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
==> ERROR: Failure while downloading http://repo.radeon.com/rocm/apt/debian/pool/main/r/rock-dkms/rock-dkms-firmware_3.5-32_all.deb
    Aborting...
Error downloading sources: rock-dkms-firmware-bin
```

It seems that the version is not the same as that from the website. Any advice?

**Comment by acxz** [comment]
Yep, Seems like they have reverted the version. Feel free to fix the URL to the correct one in the PKGBUILD and submit a PR.

**Comment by piotrv** [comment]
I changed the debian package version name in the build file and added the correct checksums, now:
  -> Extracting rock-dkms_3.5-30_all.deb with bsdtar
==> Removing existing $pkgdir/ directory...
==> Entering fakeroot environment...
==> Starting package()...
patching file dkms/Makefile
Hunk #1 FAILED at 2681.
1 out of 1 hunk FAILED -- saving rejects to file dkms/Makefile.rej
==> ERROR: A failure occurred in package().
    Aborting...

**Comment by acxz** [comment]
hmm seems like the Makefile might have had some significant changes. You try using the 3.5.1 link.

**Comment by gmaleshkov** [comment]
Hello, 
Package - rock-dkms  -http://repo.radeon.com/rocm/apt/debian/pool/main/r/rock-dkms/rock-dkms_3.5-32_all.deb file don't exists. rock-dkms_3.5-30_all.deb, but gives: Error! Bad return status for module build on kernel: 5.7.7-arch1-1 (x86_64)
Package - rock-dkms-bin - http://repo.radeon.com/rocm/apt/debian/pool/main/r/rock-dkms/rock-dkms_3.5-32_all.deb file don't exists, with  rock-dkms_3.5-30_all.deb - Hunk #1 FAILED at 2681.
1 out of 1 hunk FAILED -- saving rejects to file dkms/Makefile.rej
With regards,
Georgi

**Comment by piotrv** [comment]
> hmm, that is weird the patch was actually originally applied on 3.5-30. I suspect you might be doing something incorrect. Take a look at the previous commits of the `rock-dkms-bin/PKGBUILD` file for reference.

It seems _95.patch_ has changed as well, since
```
sha256sum 95.patch 
82b9ef05c67f538bbdf07f066e059d623db2bdb7b67a1b695dab3659601092dc  95.patch

which has not been pushed to the git repo.

Anyhow, with the corrected checksums, it seems compilation hangs within the current 95.patch, to put in my two-penny worth . .

**Comment by acxz** [comment]
I see, what if the URL is tmp changed to follow the 3.5.1 specifically? That might be worth a shot.

**Comment by piotrv** [comment]
> I see, what if the URL is tmp changed to follow the 3.5.1 specifically? That might be worth a shot.

You mean the source url to `/rocm/apt/3.5.1/pool/main/r/rock-dkms/`
and then use the available
```
rock-dkms-firmware_3.5-32_all.deb                  17-Jun-2020 19:40             4342412
rock-dkms_3.5-32_all.deb     
```

**Comment by piotrv** [comment]
I hardcoded source url  to 3.5.1

```
Building rock-dkms-bin...
==> Making package: rock-dkms-bin 3.5-4 (Sat 11 Jul 2020 22:35:47)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found rock-dkms_3.5-32_all.deb
  -> Found rock_compatibility.patch
==> Validating source files with sha256sums...
    rock-dkms_3.5-32_all.deb ... Passed
    rock_compatibility.patch ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting rock-dkms_3.5-32_all.deb with bsdtar
==> Removing existing $pkgdir/ directory...
==> Entering fakeroot environment...
==> Starting package()...
patching file dkms/Makefile
Hunk #1 FAILED at 2681.
1 out of 1 hunk FAILED -- saving rejects to file dkms/Makefile.rej
==> ERROR: A failure occurred in package().
    Aborting...
```

**Comment by acxz** [comment]
Yeah seems like upstream rewrote some of the patch for some reason. So then a couple of things we can do.

We can ship the package out without the patch. This will resolve this issue but remove the `patched` label of #267. Which I am fine doing since it's not like the patch is gonna be effective if we don't even have the sources downloaded.

The next thing will be to resolve the patch issue. We can try to use the last working version of upstream pull request 95 that @agravgaard should still have on his own branch.

I like this idea since we can solve one problem at a time. TLDR I will accept a PR that removes the patch and uses the 3.5.1 url.

**Comment by acxz** [comment]
Decided to just go ahead and fix this. Should be good with 1e0067c
